keras_cv/layers/preprocessing/erasing.py

```diff
@@ -0,0 +1,196 @@
+import abc
+
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.python.keras.utils import layer_utils
+
+from keras_cv.utils import fill_utils
+
+
+class BaseErasing(layers.Layer, abc.ABC):
```

Comment:
We should not depend on ABC for base layers (see other base layers in the Keras codebase for examples). `@abc.abstractmethod` decorators should be replaced with `NotImplementedError`.

Reply:
In this specific case I also recommend *not* having this base layer at all, instead having independent augmentation layers. Most of the shared code can be moved to stateless utility functions (e.g. `_erase` which would take a patch size computation function). Some of the shared code should be duplicated across layers to improve code readability: in particular all layers should have their own `call()` method. Explicit is better than implicit.

In terms of file structure, I recommend putting each layer of the 2 layers in its own file, named after the layer name.

Reply:
I see. The layers are now independent, and each have their own file.